core/src/main/java/com/amplifyframework/core/model/query/QueryOptions.java

```diff
@@ -60,6 +78,17 @@ public QueryOptions paginated(@NonNull final QueryPaginationInput paginationInpu
         return this;
     }
 
+    /**
+     * Updates the current options with a given {@code sortBy}.
+     *
+     * @param sortBy sorting information
+     * @return current options with an updated {@code sortBy}.
+     */
+    public QueryOptions sorted(@NonNull final QuerySortBy... sortBy) {
```

Comment:
The new `sorted` and `matches` change `QueryOptions` into a mutable class. 

One ~easy solution would be to have these generate clones, that "patch" the field of interest.

e.g.
```java
    public QueryOptions sorted(@NonNull final QuerySortBy... sortBy) {
        return new QueryOptions(queryPredicate, paginationInput, Immutable.of(Arrays.asList(sortBy)));
    }
```
Also, it would certainly be ___cool___* if we could build up the list of `QuerySortBy` through subsequent appended calls:
```java
Where.sorted(User.LAST_NAME.ascending())
    .sorted(User.FIRST_NAME.ascending())
    .sorted(User.CREATED_AT.descending())
```

If we were to support that, I guess the impl of this method would be more like:
```java
    public QueryOptions sorted(@NonNull final QuerySortBy... sortBy) {
        List<QuerySortBy> augmentedList = new ArrayList<>(this.sortBy);
        augmentedList.addAll(Immutable.of(sortBy));
        return new QueryOptions(queryPredicate, paginationInput, Immutable.of(augmentedList));
    }
```